### ISSUE TITLE
[Question] Why does the F1 score stay constant in the data hyper-cleaning example while accuracy and validation loss change?

### ISSUE THREAD

**Issue by mui369level** [callous-youth/BOAT#14]
### Required prerequisites

- [x] I have read the documentation <https://boat.readthedocs.io/en/main/>.
- [x] I have searched the [Issue Tracker](https://github.com/callous-youth/BOAT/issues) to make sure this question has not already been answered.

### Question

Environment
- OS: Linux
- Python: 3.12.12
- torch: 2.6.0+cu124
- boat-torch: 1.0.2
- Other deps (higher/torchmeta/torchvision/etc): torchvision 0.21.0+cu124 (installed), higher NOT installed,  torchmeta NOT installed

What I’m trying to do
I am trying to reproduce and compare different operator combinations in the data hyper-cleaning example for a course assignment. In my experiment, I ran three combinations: NGD+CG, DI+NGD+CG, and GDA+NGD+CG. To obtain more informative convergence curves, I increased the outer-loop iteration number to 30 and recorded the printed metrics, including accuracy, F1 score, and validation loss.

Repro steps (if applicable)
1. Clone the BOAT repository and install the required environment.
2. Run the data hyper-cleaning example with different operator combinations, for example:
   - `python examples/data_hyper_cleaning/data_hyper_cleaning.py --gm_op NGD --na_op CG`
   - `python examples/data_hyper_cleaning/data_hyper_cleaning.py --gm_op DI,NGD --na_op CG`
   - `python examples/data_hyper_cleaning/data_hyper_cleaning.py --gm_op GDA,NGD --na_op CG`
3. Increase the outer-loop iterations to 30 and plot the logged metrics.

Expected behavior
I expected the F1 score to change together with the optimization process, especially because both accuracy and validation loss change significantly across iterations and across operator combinations. I also expected the final F1 score to show some difference among NGD+CG, DI+NGD+CG, and GDA+NGD+CG.

Actual behavior
The example runs successfully, and the accuracy and validation loss change as expected. However, the F1 score appears to stay almost exactly constant at around 66.67 during the whole optimization process for all three operator combinations. For example, in my plots:
- NGD+CG: validation loss decreases and accuracy increases, but F1 remains nearly constant
- DI+NGD+CG: validation loss decreases rapidly and accuracy rises to about 88+, but F1 still remains nearly constant
- GDA+NGD+CG: validation loss decreases and accuracy rises to about 76+, but F1 again remains nearly constant

This makes me unsure whether:
1. the F1 metric is computed correctly in this example,
2. the F1 being printed is macro/micro/weighted in a way that makes it insensitive here, or
3. I may be misunderstanding how the evaluation is implemented.

Logs / traceback
The program finishes without errors. Example commands:
- `python examples/data_hyper_cleaning/data_hyper_cleaning.py --gm_op NGD --na_op CG`
- `python examples/data_hyper_cleaning/data_hyper_cleaning.py --gm_op DI,NGD --na_op CG`
- `python examples/data_hyper_cleaning/data_hyper_cleaning.py --gm_op GDA,NGD --na_op CG`

Observed results from my runs:
- NGD+CG: accuracy increases from about 10.92 to about 69.91, while val_loss decreases from about 2.311 to about 1.994, but F1 stays around 66.67
- DI+NGD+CG: accuracy increases from about 60.85 to about 88.72, while val_loss decreases from about 1.857 to about 0.398, but F1 stays around 66.67
- GDA+NGD+CG: accuracy increases from about 18.78 to about 76.16, while val_loss decreases from about 2.286 to about 1.667, but F1 stays around 66.67

Extra context / suggestion (optional)
It would be very helpful if the documentation or example code could clarify how the F1 score is computed in this task. If this is expected behavior, a short explanation would help new users understand why the F1 score remains unchanged even when other metrics improve significantly. If not, perhaps the evaluation code could be checked or documented more clearly.
